gadget/install/install.go

```diff
@@ -163,37 +170,64 @@ func installOnePartition(part *gadget.OnDiskStructure, encryptionType secboot.En
 			})
 		})
 	if err != nil {
-		return "", nil, fmt.Errorf("cannot make filesystem for partition %s: %v", partDisp, err)
+		return fmt.Errorf("cannot make filesystem for partition %s: %v", partDisp, err)
 	}
+	return nil
+}
 
+func writePartitionContent(part *gadget.OnDiskStructure, perfTimings timings.Measurer, observer gadget.ContentObserver, fsDevice string) error {
+	partDisp := roleOrLabelOrName(part)
+	var err error
 	timings.Run(perfTimings, fmt.Sprintf("write-content[%s]", partDisp),
 		fmt.Sprintf("Write content for %s", partDisp),
 		func(timings.Measurer) {
 			err = writeFilesystemContent(part, fsDevice, observer)
 		})
 	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func installOnePartition(part *gadget.OnDiskStructure, encryptionType secboot.EncryptionType, sectorSize quantity.Size, observer gadget.ContentObserver, perfTimings timings.Measurer) (fsDevice string, encryptionKey keys.EncryptionKey, err error) {
+	// 1. Encrypt
+	partDisp := roleOrLabelOrName(part)
+	fsDevice, encryptionKey, fsSectorSize, err := encryptPartition(part, encryptionType, sectorSize, perfTimings)
+	if err != nil {
+		return "", nil, fmt.Errorf("cannot encrypt partition %s: %v", partDisp, err)
+	}
+
+	// 2. Create filesystem
+	if err := createFilesystem(part, perfTimings, fsSectorSize, fsDevice); err != nil {
 		return "", nil, err
 	}
+
+	// 3. Write content
+	if err := writePartitionContent(part, perfTimings, observer, fsDevice); err != nil {
+		return "", nil, err
+	}
+
 	return fsDevice, encryptionKey, nil
 }
 
-// Run bootstraps the partitions of a device, by either creating
-// missing ones or recreating installed ones.
-func Run(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options Options, observer gadget.ContentObserver, perfTimings timings.Measurer) (*InstalledSystemSideData, error) {
+// Returns laidout volume, on disk structure, sector size, volume name from gadget.yaml
+func createPartitions(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options Options,
+	perfTimings timings.Measurer) (map[string]*gadget.LaidOutVolume, []gadget.OnDiskStructure, quantity.Size, string, error) {
```

Comment:
it seems cleaner to me to follow the patterns of the preexisting in functions an make the return values:

created, laidOutVBoolVol, allLaidOutVols, bootDiskLayout, error

Reply:
Changed now, although names are not exactly as you wrote them - I've tried to make something similar in spirit.